app/javascript/components/DateAndTimePicker.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react'
+import ReactDatePicker from 'react-datepicker'
+import classNames from 'classnames'
+import moment from 'moment'
+import { dateAndTimePickerDefaultOptions, backendDateFormat, backendHoursFormat, backendMinutesFormat } from './Constants'
+import { nl, enUS } from 'date-fns/locale'
+
+const DateAndTimePicker = props => {
+  const calculateStartDate = () => {
+    if (props.today) return moment().set({ minute: 0 }).toDate()
+    if (props.default_date) return moment(props.default_date).toDate()
+    return undefined
+  }
+  const [startDate, setStartDate] = useState(calculateStartDate())
+
+  const calculateMin = () => {
+    if (!props.min || props.min.length === 0) return undefined
+    if (props.max && (props.max === true || props.max === false)) {
+      // if min is a number
+      if (parseInt(props.min).toString() === props.min.toString()) {
+        const minNum = parseInt(props.min)
+        if (minNum > 0) return moment().add(minNum, 'days').toDate()
+        if (minNum < 0) return moment().subtract(minNum, 'days').toDate()
+        return moment().toDate()
+      }
+    }
+    return moment(props.min).toDate()
+  }
+
+  const calculateMax = () => {
+    if (!props.max || props.max.length === 0) return undefined
+    if (props.max === true) return moment().toDate()
+    return moment(props.max).toDate()
+  }
+
+  return (
+    <div className='input-field'>
+      <ReactDatePicker
+        id={props.id}
+        {...dateAndTimePickerDefaultOptions}
+        selected={startDate}
+        onChange={date => setStartDate(date)}
+        minDate={calculateMin()}
+        maxDate={calculateMax()}
+        required={props.required}
+        showTimeSelect
+        timeIntervals={15}
+        locale={(props.locale && props.locale === 'nl') ? nl : enUS}
+        timeCaption={(props.locale && props.locale === 'nl') ? 'Tijd' : 'Time'}
```

Comment:
not a big fan of this, but I think its fine, especially since we don't have i18n right now her, right?

Reply:
I think we do have i18n here. I changed it to an i18n string where we just set the locale parameter. It works.